plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMappingsProviderFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.s3;
+
+import io.airlift.log.Logger;
+
+import java.util.List;
+
+public class S3SecurityMappingsProviderFactory
+{
+    private static final Logger log = Logger.get(S3SecurityMappingsProviderFactory.class);
+
+    private S3SecurityMappingsProviderFactory()
+    {}
+
+    private static long getNumberOfMatchingProviders(S3SecurityMappingConfig config)
+    {
+        List<S3SecurityMappingsProvider> providers = List.of(new FileBasedS3SecurityMappingsProvider(config), new UriBasedS3SecurityMappingsProvider(config));
+        return providers.stream().filter(S3SecurityMappingsProvider::checkPreconditions).count();
+    }
+
+    public static S3SecurityMappingsProvider createMappingsProvider(S3SecurityMappingConfig config)
+    {
+        List<S3SecurityMappingsProvider> providers = List.of(new FileBasedS3SecurityMappingsProvider(config), new UriBasedS3SecurityMappingsProvider(config));
+        long matching = getNumberOfMatchingProviders(config);
+        if (matching == 0) {
+            throw new IllegalArgumentException("No provider available for current hive.s3.security-mapping configuration. " +
```

Comment:
Sorry, I do not fully understand your comment. `isProivderAvailable()` should return a boolean indicating if a single, unambiguous provider can be created. Of course, we cannot force devs to call this method first before calling `createMappingsProvider(...)`. Here I wanted to distinguish between ambigous configuration and no provider configured at all as different error causes.

Reply:
I guess it does not matter anyways, because I removed `S3SecurityMappingsProviderFactory.java` completely in favour of setup via airlift & binders as you suggested in your comment about using airlift's HttpClient.